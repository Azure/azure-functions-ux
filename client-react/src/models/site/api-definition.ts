export interface ApiDefinition {
  url: string;
}
