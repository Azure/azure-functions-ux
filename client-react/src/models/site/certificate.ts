import { HostingEnvironmentProfile } from '../hostingEnvironment/hosting-environment-profile';
import { KeyValue } from '../portal-models';

export interface Certificate {
  friendlyName: string;
  subjectName: string;
  hostNames: string;
  pfxBlob: number;
  siteName: string;
  selfLink: string;
  issuer: string;
  issueDate: Date;
  expirationDate: Date;
  password: string;
  thumbprint: string;
  valid: boolean;
  toDelete: boolean;
  cerBlob: number;
  publicKeyHash: string;
  hostingEnvironment: string;
  hostingEnvironmentProfile: HostingEnvironmentProfile;
  keyVaultCsmId: string;
  keyVaultSecretName: string;
  webSpace: string;
  serverFarmId: string;
  geoRegion: string;
  name: string;
  tags: KeyValue<string>;
}

export interface Csr {
  name: string;
  distinguishedName: string;
  csrString: string;
  pfxBlob: number;
  password: string;
  publicKeyHash: string;
  hostingEnvironment: string;
}
