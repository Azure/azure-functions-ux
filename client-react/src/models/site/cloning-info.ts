import { KeyValue } from '../portal-models';

export interface CloningInfo {
  correlationId: string;
  overwrite: boolean;
  cloneCustomHostNames: boolean;
  cloneSourceControl: boolean;
  sourceWebAppId: string;
  hostingEnvironment: string;
  appSettingsOverrides: KeyValue<string>;
  configureLoadBalancing: boolean;
  trafficManagerProfileId: string;
  trafficManagerProfileName: string;
  ignoreQuotas: boolean;
}
