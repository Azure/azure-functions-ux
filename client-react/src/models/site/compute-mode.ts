export enum ComputeMode {
  Shared = 0,
  Dedicated,
  Dynamic,
}
