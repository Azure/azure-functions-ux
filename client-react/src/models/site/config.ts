import { NameValuePair } from '../name-value-pair';
import { ApiDefinition } from './api-definition';

export enum ScmType {
  None = 'None',
  Dropbox = 'Dropbox',
  Tfs = 'Tfs',
  LocalGit = 'LocalGit',
  GitHub = 'GitHub',
  GitHubAction = 'GitHubAction',
  CodePlexGit = 'CodePlexGit',
  CodePlexHg = 'CodePlexHg',
  BitbucketGit = 'BitbucketGit',
  BitbucketHg = 'BitbucketHg',
  ExternalGit = 'ExternalGit',
  ExternalHg = 'ExternalHg',
  OneDrive = 'OneDrive',
  Vso = 'VSO',
  Vsts = 'VSTS',
}

export enum BuildProvider {
  None = 'None',
  GitHubAction = 'GitHubAction',
  AppServiceBuildService = 'AppServiceBuildService',
}

export interface SiteConfig {
  numberOfWorkers: number;
  defaultDocuments: string[];
  netFrameworkVersion: string;
  phpVersion: string;
  pythonVersion: string;
  nodeVersion: string;
  windowsFxVersion: string;
  linuxFxVersion: string;
  minTlsVersion: string;
  requestTracingEnabled: boolean;
  requestTracingExpirationTime: Date;
  remoteDebuggingEnabled: boolean;
  remoteDebuggingVersion: string | null;
  httpLoggingEnabled: boolean;
  logsDirectorySizeLimit: number;
  detailedErrorLoggingEnabled: boolean;
  publishingUsername: string;
  publishingPassword: string;
  push: PushSettings;
  appSettings: NameValuePair[];
  metadata: NameValuePair[];
  connectionStrings: ConnStringInfo[];
  handlerMappings: HandlerMapping[];
  azureStorageAccounts?: ArmAzureStorageMount;
  documentRoot: string;
  scmType: ScmType;
  use32BitWorkerProcess: boolean;
  webSocketsEnabled: boolean;
  alwaysOn: boolean;
  javaVersion: string;
  javaContainer: string;
  javaContainerVersion: string;
  managedPipelineMode: number;
  ftpsState?: 'AllAllowed' | 'FtpsOnly' | 'Disabled';
  http20Enabled?: boolean;
  virtualApplications: VirtualApplication[];
  winAuthAdminState: number;
  winAuthTenantState: number;
  customAppPoolIdentityAdminState: boolean;
  customAppPoolIdentityTenantState: boolean;
  runtimeADUser: string;
  runtimeADUserPassword: string;
  loadBalancing: number;
  routingRules: RoutingRule[];
  experiments: Experiments;
  limits: SiteLimits;
  autoHealEnabled: boolean;
  autoHealRules: AutoHealRules;
  tracingOptions: string;
  vnetName: string;
  siteAuthEnabled: boolean;
  siteAuthSettings: SiteAuthSettings;
  autoSwapSlotName: string;
  apiDefinition: ApiDefinition;
  cors: Cors;
  localMySqlEnabled: boolean;
  appCommandLine: string;
  ipSecurityRestrictions: IpRestriction[];
  scmIpSecurityRestrictions: IpRestriction[];
  reservedInstanceCount?: number;
  functionsRuntimeScaleMonitoringEnabled?: boolean;
}

export interface IpRestriction {
  ipAddress: string;
  action: string;
  tag: string;
  priority: number;
  name: string;
  description: string;
}

export interface PushSettings {
  isPushEnabled: boolean;
  tagWhitelistJson: string;
  tagsRequiringAuth: string;
  dynamicTagsJson: string;
}

export interface ConnStringInfo {
  name: string;
  connectionString: string;
  type: string;
}

export interface HandlerMapping {
  extension: string;
  scriptProcessor: string;
  arguments: string;
}

export interface AzureStorageMount {
  type: StorageType;
  accountName: string;
  shareName: string;
  accessKey: string;
  mountPath: string;
}

export enum StorageType {
  azureFiles = 'AzureFiles',
  azureBlob = 'AzureBlob',
}

export interface ArmAzureStorageMount {
  [key: string]: AzureStorageMount;
}

export interface VirtualApplication {
  virtualPath: string;
  physicalPath: string;
  virtualDirectory: boolean;
  preloadEnabled?: boolean;
  virtualDirectories?: VirtualDirectory[];
}

export interface VirtualDirectory {
  virtualPath: string;
  physicalPath: string;
}

export interface RoutingRule {
  name: string;
}

export interface Experiments {
  rampUpRules: RampUpRule;
}

export interface RampUpRule {
  actionHostName: string;
  reroutePercentage: number;
  changeStep: number;
  changeIntervalInMinutes: number;
  minReroutePercentage: number;
  maxReroutePercentage: number;
  changeDecisionCallbackUrl: string;
  name: string;
}

export interface SiteLimits {
  maxPercentageCpu: number;
  maxMemoryInMb: number;
  maxDiskSizeInMb: number;
}

export interface AutoHealTriggers {
  privateBytesInKB: number;
}

export interface AutoHealCustomAction {
  exe: string;
  parameters: string;
}

export interface AutoHealActions {
  actionType: number;
  customAction: AutoHealCustomAction;
}

export interface AutoHealRules {
  triggers: AutoHealTriggers;
  actions: AutoHealActions;
}

export interface SiteAuthSettings {
  enabled: boolean;
  httpApiPrefixPath: string;
  unauthenticatedClientAction: number;
  tokenStoreEnabled: boolean;
  allowedExternalRedirectUrls: string;
  allowedAudiences: string;
  defaultProvider: number;
  legacyMobileCompatibilityMode: boolean;
  appObjectId: string;
  clientId: string;
  clientSecret: string;
  issuer: string;
  validateIssuer: boolean;
  isAadAutoProvisioned: boolean;
  googleClientId: string;
  googleClientSecret: string;
  googleOAuthScopes: string;
  facebookAppId: string;
  facebookAppSecret: string;
  facebookOAuthScopes: string;
  twitterConsumerKey: string;
  twitterConsumerSecret: string;
  microsoftAccountClientId: string;
  microsoftAccountClientSecret: string;
  microsoftAccountOAuthScopes: string;
}

export interface Cors {
  allowedOrigins: string;
}

export interface KeyVaultReference {
  reference: string;
  status: string;
  vaultName?: string;
  secretName?: string;
  secretVersion?: string;
  identityType?: string;
  details: string;
  source?: string;
  location?: 'ApplicationSetting' | 'ConnectionString';
}
