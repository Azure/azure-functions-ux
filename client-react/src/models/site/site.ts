import { SiteProperties } from './site-properties';
import { HostingEnvironmentProfile } from '../hostingEnvironment/hosting-environment-profile';
import { Certificate, Csr } from './certificate';
import { SiteConfig } from './config';
import { CloningInfo } from './cloning-info';
import { KeyValue } from '../portal-models';

export enum ContentAvailabilityState {
  Normal = 'Normal',
  ReadOnly = 'ReadOnly',
}

export enum RuntimeAvailabilityState {
  Normal = 'Normal',
  Degraded = 'Degraded',
  NotAvailable = 'NotAvailable',
}

export enum SiteAvailabilityState {
  Normal = 'Normal',
  Limited = 'Limited',
  DisasterRecoveryMode = 'DisasterRecoveryMode',
}

export enum UsageState {
  Normal = 'Normal',
  Exceeded = 'Exceeded',
}

export enum SslState {
  Disabled = 'Disabled',
  SniEnabled = 'SniEnabled',
  IpBasedEnabled = 'IpBasedEnabled',
}

export enum HostType {
  Standard = 'Standard',
  Repository = 'Repository',
}

export enum IpBasedSslState {
  NotConfigured = 'NotConfigured',
  InProgress = 'InProgress',
  Configured = 'Configured',
  ConfigurationReverted = 'ConfigurationReverted',
}

export enum SiteDisabledReason {
  Undefined = 0,
  FunctionQuotaExceeded = 1,
  Other = 100,
}

export enum ClientCertMode {
  Required = 'Required',
  Optional = 'Optional',
}

export interface Site {
  name: string;
  state: string;
  defaultHostName: string;
  hostNames: string[];
  webSpace: string;
  selfLink: string;
  repositorySiteName: string;
  owner: string;
  usageState: UsageState;
  enabled: boolean;
  adminEnabled: boolean;
  enabledHostNames: string[];
  siteProperties: Partial<SiteProperties>;
  availabilityState: SiteAvailabilityState;
  sSLCertificates: Certificate;
  csrs: Csr[];
  cers: Certificate;
  siteMode: string;
  hostNameSslStates: Partial<HostNameSslState>[];
  computeMode: number;
  serverFarm: string;
  serverFarmId: string;
  lastModifiedTimeUtc: string;
  storageRecoveryDefaultState: string;
  contentAvailabilityState: ContentAvailabilityState;
  runtimeAvailabilityState: RuntimeAvailabilityState;
  siteConfig: Partial<SiteConfig>;
  deploymentId: string;
  trafficManagerHostNames: string;
  sku: string;
  premiumAppDeployed: boolean;
  scmSiteAlsoStopped: boolean;
  targetSwapSlot: string;
  hostingEnvironmentProfile?: HostingEnvironmentProfile;
  microService: string;
  gatewaySiteName: string;
  clientAffinityEnabled: boolean;
  clientCertEnabled: boolean;
  clientCertMode: ClientCertMode;
  clientCertExclusionPaths: string;
  hostNamesDisabled: boolean;
  domainVerificationIdentifiers: string;
  kind: string;
  httpsOnly: boolean;
  outboundIpAddresses: string;
  geoRegion: string;
  cloningInfo: CloningInfo;
  hostingEnvironmentId: string;
  tags: KeyValue<string>;
  resourceGroup: string;
  isLinux: boolean;
  hyperV: boolean;
  reserved: boolean;
  dailyMemoryTimeQuota: number;
  siteDisabledReason: SiteDisabledReason;
  possibleInboundIpAddresses?: string;
}

export interface HostNameSslState {
  name: string;
  sslState: SslState;
  ipBasedSslResult: string;
  virtualIP: string;
  thumbprint: string;
  toUpdate: boolean;
  toUpdateIpBasedSsl: boolean;
  iPBasedSslState: IpBasedSslState;
  hostType: HostType;
}

export enum AppOs {
  linux = 'linux',
  windows = 'windows',
}

export type AppOsType = AppOs.linux | AppOs.windows;
