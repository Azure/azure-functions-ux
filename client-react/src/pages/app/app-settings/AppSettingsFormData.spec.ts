import {
  convertStateToForm,
  getFormAppSetting,
  getFormConnectionStrings,
  flattenVirtualApplicationsList,
  unFlattenVirtualApplicationsList,
  getCurrentStackString,
  convertFormToState,
  getConfigWithStackSettings,
} from './AppSettingsFormData';
import { mockSite, mockWebConfig, mockConnectionStrings, mockMetadata, mockAppSettings, mockSlotConfigName } from '../../../mocks/ArmMocks';
import { VirtualApplication } from '../../../models/site/config';

describe('Convert State to Form Data', () => {
  it('convert redux state to form state', () => {
    const mockFormData = convertStateToForm(mockProps as any);
    expect(mockFormData.virtualApplications.length).toBe(3);
    expect(mockFormData.appSettings.length).toBe(2);
    expect(mockFormData.connectionStrings.length).toBe(1);
    expect(mockFormData.currentlySelectedStack).toBeTruthy();
  });
});

describe('Get Form App Setting', () => {
  it('Converts app settings to app settings with slot config tick', () => {
    const mockAppSettingsForm = getFormAppSetting(mockProps.appSettings, mockProps.slotConfigNames);
    expect(mockAppSettingsForm.length).toBe(2);
    expect(mockAppSettingsForm.filter(x => x.sticky).length).toBe(1);
  });
});

describe('Get Arm App Setting Objects', () => {
  it('Converts app settings to app settings with slot config tick', () => {
    const mockFormData = convertStateToForm(mockProps as any);
    const newObjs = convertFormToState(mockFormData, mockProps.metadata, mockProps.slotConfigNames);
    expect(newObjs).toMatchSnapshot();
  });
});

describe('Get Form Connection Strings', () => {
  it('Converts connection strings to connection strings with slot config tick', () => {
    const mockConnectionStringsForm = getFormConnectionStrings(mockProps.connectionStrings, mockProps.slotConfigNames);
    expect(mockConnectionStringsForm.length).toBe(1);
    expect(mockConnectionStringsForm.filter(x => x.sticky).length).toBe(1);
  });
});

describe('Flatten Virtual Applications List', () => {
  it('converts API virtual application list to form virtual application list', () => {
    const mockVirtualApplicationForm = flattenVirtualApplicationsList(mockProps.config.properties
      .virtualApplications as VirtualApplication[]);
    expect(mockVirtualApplicationForm.length).toBe(3);
    expect(mockVirtualApplicationForm.filter(x => x.virtualDirectory).length).toBe(1);
    expect(mockVirtualApplicationForm.filter(x => !x.virtualDirectory).length).toBe(2);
  });

  it('null api value returns empty array', () => {
    const mockVirtualApplicationForm = flattenVirtualApplicationsList(null);
    expect(mockVirtualApplicationForm.length).toBe(0);
  });
});

describe('Unflatten Virtual Applications List', () => {
  it('converts Form virtual application list back to api virtual application list', () => {
    const mockVirtualApplicationForm = flattenVirtualApplicationsList(mockProps.config.properties
      .virtualApplications as VirtualApplication[]);
    const mockVirtualApplicationApi = unFlattenVirtualApplicationsList(mockVirtualApplicationForm);
    expect(mockVirtualApplicationApi.length).toBe(2);
    expect(mockVirtualApplicationApi[0].virtualDirectories!.length).toBe(1);
  });

  it("Handles virtual paths that don't start with /", () => {
    const virtualApplicationsFormData = [
      {
        virtualPath: '/',
        physicalPath: 'site\\wwwroot',
        preloadEnabled: true,
        virtualDirectory: false,
      },
      {
        virtualPath: '/testapp',
        physicalPath: 'site\\testapp',
        preloadEnabled: false,
        virtualDirectory: false,
        virtualDirectories: [],
      },
      {
        virtualPath: 'test2',
        physicalPath: 'site\\wwwroot\\dir',
        virtualDirectory: true,
      },
    ];
    const expectedVirtualApplicationsApiData = [
      { physicalPath: 'site\\testapp', preloadEnabled: false, virtualDirectories: [], virtualDirectory: false, virtualPath: '/testapp' },
      {
        physicalPath: 'site\\wwwroot',
        preloadEnabled: true,
        virtualDirectories: [{ physicalPath: 'site\\wwwroot\\dir', virtualDirectory: true, virtualPath: 'test2' }],
        virtualDirectory: false,
        virtualPath: '/',
      },
    ];
    const unflattenedData = unFlattenVirtualApplicationsList(virtualApplicationsFormData);
    expect(unflattenedData).toEqual(expectedVirtualApplicationsApiData);
  });

  it('Handles virtual paths that does start with /', () => {
    const virtualApplicationsFormData = [
      {
        virtualPath: '/',
        physicalPath: 'site\\wwwroot',
        preloadEnabled: true,
        virtualDirectory: false,
      },
      {
        virtualPath: '/testapp',
        physicalPath: 'site\\testapp',
        preloadEnabled: false,
        virtualDirectory: false,
        virtualDirectories: [],
      },
      {
        virtualPath: '/test2',
        physicalPath: 'site\\wwwroot\\dir',
        virtualDirectory: true,
      },
    ];
    const expectedVirtualApplicationsApiData = [
      { physicalPath: 'site\\testapp', preloadEnabled: false, virtualDirectories: [], virtualDirectory: false, virtualPath: '/testapp' },
      {
        physicalPath: 'site\\wwwroot',
        preloadEnabled: true,
        virtualDirectories: [{ physicalPath: 'site\\wwwroot\\dir', virtualDirectory: true, virtualPath: 'test2' }],
        virtualDirectory: false,
        virtualPath: '/',
      },
    ];
    const unflattenedData = unFlattenVirtualApplicationsList(virtualApplicationsFormData);
    expect(unflattenedData).toEqual(expectedVirtualApplicationsApiData);
  });

  it('Handles virtual paths that has mixed / usage', () => {
    const virtualApplicationsFormData = [
      {
        virtualPath: '/',
        physicalPath: 'site\\wwwroot',
        preloadEnabled: true,
        virtualDirectory: false,
      },
      {
        virtualPath: '/testapp',
        physicalPath: 'site\\testapp',
        preloadEnabled: false,
        virtualDirectory: false,
        virtualDirectories: [],
      },
      {
        virtualPath: '/test2',
        physicalPath: 'site\\wwwroot\\dir',
        virtualDirectory: true,
      },
      {
        virtualPath: 'test3',
        physicalPath: 'site\\wwwroot\\dir',
        virtualDirectory: true,
      },
    ];
    const expectedVirtualApplicationsApiData = [
      { physicalPath: 'site\\testapp', preloadEnabled: false, virtualDirectories: [], virtualDirectory: false, virtualPath: '/testapp' },
      {
        physicalPath: 'site\\wwwroot',
        preloadEnabled: true,
        virtualDirectories: [
          { physicalPath: 'site\\wwwroot\\dir', virtualDirectory: true, virtualPath: 'test2' },
          { physicalPath: 'site\\wwwroot\\dir', virtualDirectory: true, virtualPath: 'test3' },
        ],
        virtualDirectory: false,
        virtualPath: '/',
      },
    ];
    const unflattenedData = unFlattenVirtualApplicationsList(virtualApplicationsFormData);
    expect(unflattenedData).toEqual(expectedVirtualApplicationsApiData);
  });
});

describe('Get Current Stack', () => {
  it('returns java if java version is there', () => {
    const configData = { ...mockProps.config } as any;
    const metadata = { ...mockProps.metadata };
    configData.properties.javaVersion = '1.8';
    const currentStack = getCurrentStackString(configData, metadata);
    expect(currentStack).toBe('java');
  });

  it('returns .net as default is nothing else is there', () => {
    const configData = { ...mockProps.config } as any;
    const metadata = { ...mockProps.metadata };
    configData.properties.javaVersion = null;
    const currentStack = getCurrentStackString(configData, metadata);
    expect(currentStack).toBe('dotnet');
  });

  it('returns what is stored in metadata absent a java version', () => {
    const configData = { ...mockProps.config } as any;
    const metadata = { ...mockProps.metadata };
    configData.properties.javaVersion = null;
    metadata.properties['CURRENT_STACK'] = 'python';
    const currentStack = getCurrentStackString(configData, metadata);
    expect(currentStack).toBe('python');
  });

  it('java version takes precidense over metadata', () => {
    const configData = { ...mockProps.config } as any;
    const metadata = { ...mockProps.metadata };
    configData.properties.javaVersion = '1.8';
    metadata.properties['CURRENT_STACK'] = 'python';
    const currentStack = getCurrentStackString(configData, metadata);
    expect(currentStack).toBe('java');
  });
});

describe('getConfigWithStackSettings', () => {
  it("removes java settings if currentlySelectedStack isn't java", () => {
    const configData = {
      ...mockProps.config,
      properties: {
        javaContainer: 'test',
        javaContainerVersion: 'test',
        javaVersion: 'test',
      },
    } as any;
    const values = { currentlySelectedStack: 'notjava' } as any;
    const configBack = getConfigWithStackSettings(configData, values);
    expect(configBack.properties.javaContainer).toBe('');
    expect(configBack.properties.javaContainerVersion).toBe('');
    expect(configBack.properties.javaVersion).toBe('');
  });
  it('leaves java settings if currently selected stack is java', () => {
    const configData = {
      ...mockProps.config,
      properties: {
        javaContainer: 'test',
        javaContainerVersion: 'test',
        javaVersion: 'test',
      },
    } as any;
    const values = { currentlySelectedStack: 'java' } as any;
    const configBack = getConfigWithStackSettings(configData, values);
    expect(configBack.properties.javaContainer).toBe('test');
    expect(configBack.properties.javaContainerVersion).toBe('test');
    expect(configBack.properties.javaVersion).toBe('test');
  });
});

const mockProps = {
  site: mockSite,
  config: mockWebConfig,
  appSettings: mockAppSettings,
  connectionStrings: mockConnectionStrings,
  metadata: mockMetadata,
  slotConfigNames: mockSlotConfigName,
};
