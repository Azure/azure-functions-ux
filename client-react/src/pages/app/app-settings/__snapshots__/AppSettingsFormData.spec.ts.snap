// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get Arm App Setting Objects Converts app settings to app settings with slot config tick 1`] = `
Object {
  "site": Object {
    "id": "/subscriptions/sub/resourceGroups/azure-functions-react-test/providers/Microsoft.Web/sites/azure-functions-react-test",
    "kind": "app,linux",
    "location": "Central US",
    "name": "azure-functions-react-test",
    "properties": Object {
      "adminEnabled": true,
      "availabilityState": "Normal",
      "cers": null,
      "clientAffinityEnabled": true,
      "clientCertEnabled": false,
      "cloningInfo": null,
      "computeMode": null,
      "containerSize": 0,
      "contentAvailabilityState": "Normal",
      "csrs": Array [],
      "dailyMemoryTimeQuota": 0,
      "defaultHostName": "azure-functions-react-test.azurewebsites.net",
      "deploymentId": "azure-functions-react-test",
      "domainVerificationIdentifiers": null,
      "enabled": true,
      "enabledHostNames": Array [
        "azure-functions-react-test.azurewebsites.net",
        "azure-functions-react-test.scm.azurewebsites.net",
      ],
      "functionExecutionUnitsCache": null,
      "homeStamp": "waws-prod-dm1-027",
      "hostNameSslStates": Array [
        Object {
          "hostType": "Standard",
          "ipBasedSslResult": null,
          "ipBasedSslState": "NotConfigured",
          "name": "azure-functions-react-test.azurewebsites.net",
          "sslState": "Disabled",
          "thumbprint": null,
          "toUpdate": null,
          "toUpdateIpBasedSsl": null,
          "virtualIP": null,
        },
        Object {
          "hostType": "Repository",
          "ipBasedSslResult": null,
          "ipBasedSslState": "NotConfigured",
          "name": "azure-functions-react-test.scm.azurewebsites.net",
          "sslState": "Disabled",
          "thumbprint": null,
          "toUpdate": null,
          "toUpdateIpBasedSsl": null,
          "virtualIP": null,
        },
      ],
      "hostNames": Array [
        "azure-functions-react-test.azurewebsites.net",
      ],
      "hostNamesDisabled": false,
      "hostingEnvironment": null,
      "hostingEnvironmentProfile": null,
      "hostingEnvironmentid": "",
      "httpsOnly": false,
      "isXenon": false,
      "kind": "app",
      "lastModifiedTimeUtc": "2018-10-17T17:34:08.6533333",
      "maxNumberOfWorkers": null,
      "name": "azure-functions-react-test",
      "outboundIpAddresses": "52.173.249.137,52.176.59.177,52.173.201.190,52.173.203.66,52.173.251.28",
      "owner": null,
      "possibleOutboundIpAddresses": "52.173.249.137,52.176.59.177,52.173.201.190,52.173.203.66,52.173.251.28,52.173.202.202,52.173.200.111",
      "repositorySiteName": "azure-functions-react-test",
      "reserved": false,
      "resourceGroup": "azure-functions-react-test",
      "runtimeAvailabilityState": "Normal",
      "scmSiteAlsoStopped": false,
      "selfLink": "https://waws-prod-dm1-027.api.azurewebsites.windows.net:454/subscriptions/sub/webspaces/azure-functions-react-test-CentralUSwebspace/sites/azure-functions-react-test",
      "serverFarm": null,
      "serverFarmId": "/subscriptions/sub/resourceGroups/azure-functions-react-test/providers/Microsoft.Web/serverfarms/ServicePlan9131048f-9b79",
      "siteConfig": Object {
        "ClusteringEnabled": false,
        "alwaysOn": false,
        "apiDefinition": null,
        "appCommandLine": "",
        "appSettings": Array [
          Object {
            "name": "MSDEPLOY_RENAME_LOCKED_FILES",
            "value": "1",
          },
          Object {
            "name": "TEST_NAME",
            "value": "TEST_VALUE",
          },
        ],
        "autoHealEnabled": false,
        "autoHealRules": null,
        "autoSwapSlotName": null,
        "azureStorageAccounts": Object {},
        "connectionStrings": Array [
          Object {
            "connectionString": "TEST_CONNECTION_VALUE",
            "name": "TEST_CONNECTION",
            "type": "MySql",
          },
        ],
        "cors": null,
        "customAppPoolIdentityAdminState": false,
        "customAppPoolIdentityTenantState": false,
        "defaultDocuments": Array [
          "Default.htm",
          "Default.html",
          "Default.asp",
          "index.htm",
          "index.html",
          "iisstart.htm",
          "default.aspx",
          "index.php",
          "hostingstart.html",
        ],
        "detailedErrorLoggingEnabled": false,
        "documentRoot": null,
        "experiments": Object {
          "rampUpRules": Array [],
        },
        "ftpsState": "AllAllowed",
        "handlerMappings": Array [
          Object {
            "arguments": "asdasdasd",
            "extension": "sdasdasdas",
            "scriptProcessor": "asdasdasda",
          },
        ],
        "http20Enabled": false,
        "httpLoggingEnabled": false,
        "ipSecurityRestrictions": null,
        "javaContainer": "",
        "javaContainerVersion": "",
        "javaVersion": "",
        "limits": null,
        "linuxFxVersion": "",
        "loadBalancing": "LeastRequests",
        "localMySqlEnabled": false,
        "logsDirectorySizeLimit": 35,
        "machineKey": null,
        "managedPipelineMode": 0,
        "managedServiceIdentityid": "",
        "metadata": Array [
          Object {
            "name": "CURRENT_STACK",
            "value": "",
          },
        ],
        "minTlsVersion": "1.2",
        "netFrameworkVersion": "v4.0",
        "nodeVersion": null,
        "numberOfWorkers": 1,
        "phpVersion": "5.6",
        "publishingPassword": null,
        "publishingUsername": "$azure-functions-react-test",
        "push": null,
        "pythonVersion": "",
        "remoteDebuggingEnabled": false,
        "remoteDebuggingVersion": "VS2012",
        "requestTracingEnabled": false,
        "reservedInstanceCount": 0,
        "routingRules": Array [],
        "runtimeADUser": null,
        "runtimeADUserPassword": null,
        "scmType": "VSTSRM",
        "siteAuthEnabled": false,
        "siteAuthSettings": Object {
          "additionalLoginParams": null,
          "allowedAudiences": null,
          "allowedExternalRedirectUrls": null,
          "clientSecret": null,
          "clientid": "",
          "defaultProvider": null,
          "enabled": null,
          "facebookAppSecret": null,
          "facebookAppid": "",
          "facebookOAuthScopes": null,
          "googleClientSecret": null,
          "googleClientid": "",
          "googleOAuthScopes": null,
          "isAadAutoProvisioned": false,
          "issuer": null,
          "microsoftAccountClientSecret": null,
          "microsoftAccountClientid": "",
          "microsoftAccountOAuthScopes": null,
          "tokenStoreEnabled": null,
          "twitterConsumerKey": null,
          "twitterConsumerSecret": null,
          "unauthenticatedClientAction": null,
        },
        "tracingOptions": null,
        "use32BitWorkerProcess": true,
        "virtualApplications": Array [
          Object {
            "physicalPath": "site\\\\testapp",
            "preloadEnabled": false,
            "virtualDirectories": Array [
              Object {
                "physicalPath": "site\\\\wwwroot\\\\dir",
                "virtualDirectories": Array [],
                "virtualDirectory": true,
                "virtualPath": "/test2",
              },
            ],
            "virtualDirectory": false,
            "virtualPath": "/testapp",
          },
          Object {
            "physicalPath": "site\\\\wwwroot",
            "preloadEnabled": true,
            "virtualDirectories": Array [],
            "virtualDirectory": false,
            "virtualPath": "/",
          },
        ],
        "vnetName": "",
        "webSocketsEnabled": false,
        "winAuthAdminState": 0,
        "winAuthTenantState": 0,
        "windowsFxVersion": null,
        "xManagedServiceIdentityid": "",
      },
      "siteDisabledReason": 0,
      "siteMode": null,
      "siteProperties": Object {
        "appSettings": null,
        "metadata": null,
        "properties": Array [
          Object {
            "name": "LinuxFxVersion",
            "value": "TOMCAT|8.5-jre8",
          },
          Object {
            "name": "WindowsFxVersion",
            "value": null,
          },
        ],
      },
      "sku": "Standard",
      "slotSwapStatus": null,
      "sslCertificates": null,
      "state": "Running",
      "storageRecoveryDefaultState": "Running",
      "suspendedTill": null,
      "tags": null,
      "targetSwapSlot": null,
      "trafficManagerHostNames": null,
      "usageState": "Normal",
      "webSpace": "azure-functions-react-test-CentralUSwebspace",
    },
    "type": "Microsoft.Web/sites",
  },
  "slotConfigNames": Object {
    "id": "/subscriptions/sub/resourceGroups/azure-functions-react-test/providers/Microsoft.Web/sites/azure-functions-react-test/config/slotconfignames",
    "location": "Central US",
    "name": "slotconfignames",
    "properties": Object {
      "appSettingNames": Array [
        "TEST_NAME",
      ],
      "azureStorageConfigNames": Array [],
      "connectionStringNames": Array [
        "TEST_CONNECTION",
      ],
    },
  },
  "slotConfigNamesModified": true,
}
`;
